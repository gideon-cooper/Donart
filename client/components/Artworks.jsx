import React, { useState, useEffect } from 'react'
import { Link } from 'react-router-dom'
import { getArt } from '../api'
import Footer from './Footer'
import ArtworkItem from './ArtworkItem'

export default function Artworks() {
  const [artworks, setArtworks] = useState({
    artworks: [],
  })
  useEffect(() => {
    getArt()
      .then((res) => {
        setArtworks(res)
      })
      .catch((error) => {
        console.log('error: ', error.message)
      })
  }, [])
  return (
    <>
      <div className="artworkDisplay" style={{paddingBottom: "150px"}}>
        <div className="categoryTitle">
          <hr />
          <h2>Browse Artworks</h2>
          <hr />
        </div>
        <div className="card-flex-wrapper column is-three-quarters">
          {artworks.artworks.map((artwork) => (
            <ArtworkItem key={artwork.id} artwork={artwork} />
          ))}
        </div>
      </div>
      <Footer />
    </>
  )
}

// Display picture of artwork
// Display name of artwork
// display price of artwork
// Display charity artwork is going towards
