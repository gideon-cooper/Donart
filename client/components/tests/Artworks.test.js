import React from 'react'
import { HashRouter as Router } from 'react-router-dom'
import renderer from 'react-test-renderer'

import Artworks from '../Artworks'

it('renders correctly', () => {
  const tree = renderer
    .create(
      <Router>
        <Artworks />
      </Router>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
