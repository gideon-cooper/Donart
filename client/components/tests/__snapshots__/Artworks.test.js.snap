// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <div
    className="artworkDisplay"
    style={
      Object {
        "paddingBottom": "150px",
      }
    }
  >
    <div
      className="categoryTitle"
    >
      <hr />
      <h2>
        Browse Artworks
      </h2>
      <hr />
    </div>
    <div
      className="card-flex-wrapper column is-three-quarters"
    />
  </div>,
  <div
    className="footer"
  >
    <a
      href="#/About"
      onClick={[Function]}
    >
      <div
        className="firstColumn"
      >
        <h1>
          About us
        </h1>
      </div>
    </a>
    <div
      className="secondColumn"
    >
      <div>
        <h1>
          Contact us
        </h1>
        <br />
      </div>
      <div>
        <h1
          className="gideonsLineBreaks"
        >
          Copyright © 2020 – Donart® All Rights Reserved
          <br />
          Terms of Use | Privacy Policy | Disclaimer
        </h1>
      </div>
    </div>
    <div
      className="thirdColumn"
    >
      <h1>
        Legal
      </h1>
    </div>
  </div>,
]
`;
